mods/fire/init.lua

```diff
@@ -23,46 +36,84 @@ end
 
 -- Flame nodes
 
-minetest.register_node("fire:basic_flame", {
-	drawtype = "firelike",
-	tiles = {
-		{
-			name = "fire_basic_flame_animated.png",
-			animation = {
-				type = "vertical_frames",
-				aspect_w = 16,
-				aspect_h = 16,
-				length = 1
+if fire_enabled then
+	minetest.register_node("fire:basic_flame", {
+		drawtype = "firelike",
+		tiles = {
+			{
+				name = "fire_basic_flame_animated.png",
+				animation = {
+					type = "vertical_frames",
+					aspect_w = 16,
+					aspect_h = 16,
+					length = 1
+				},
 			},
 		},
-	},
-	inventory_image = "fire_basic_flame.png",
-	paramtype = "light",
-	light_source = 13,
-	walkable = false,
-	buildable_to = true,
-	sunlight_propagates = true,
-	floodable = true,
-	damage_per_second = 4,
-	groups = {igniter = 2, dig_immediate = 3, not_in_creative_inventory = 1},
-	drop = "",
+		inventory_image = "fire_basic_flame.png",
+		paramtype = "light",
+		light_source = 13,
+		walkable = false,
+		buildable_to = true,
+		sunlight_propagates = true,
+		floodable = true,
+		damage_per_second = 4,
+		groups = {igniter = 2, dig_immediate = 3, not_in_creative_inventory = 1},
+		drop = "",
+
+		on_timer = function(pos)
+			local f = minetest.find_node_near(pos, 1, {"group:flammable"})
+			if not f then
+				minetest.remove_node(pos)
+				return
+			end
+			-- Restart timer
+			return true
+		end,
+
+		on_construct = function(pos)
+			minetest.get_node_timer(pos):start(math.random(30, 60))
```

Comment:
Heh ok i almost did this but thought it wouldn't work (should have checked). I thought that the setting would not be available within the functions.

Reply:
It's a local variable, defined on load. So it will be available to the entire code in `init.lua` after its definition - it does not expire until the Lua instance is closed.